src/olympia/bandwagon/tests/test_views.py

```diff
@@ -1290,3 +1291,169 @@ def test_honeypot_statsd_incr(self, mock_incr):
+        other_url = reverse('collection-list',
+                            kwargs={'user_pk': random_user.pk})
+        collection_factory(author=random_user)
+
+        self.client.login_api(self.user)
+        response = self.client.get(other_url)
+        assert response.status_code == 403
+
+    def test_admin(self):
+        random_user = user_factory()
+        other_url = reverse('collection-list',
+                            kwargs={'user_pk': random_user.pk})
+        collection_factory(author=random_user)
+
+        self.grant_permission(self.user, 'Users:Edit')
+        self.client.login_api(self.user)
+        response = self.client.get(other_url)
+        assert response.status_code == 200
+        assert len(response.data['results']) == 1
+
+    def test_disallowed_verbs(self):
+        response = self.client.post(self.url)
+        assert response.status_code == 405
+        response = self.client.put(self.url)
+        assert response.status_code == 405
+        response = self.client.patch(self.url)
+        assert response.status_code == 405
+
+
+class TestCollectionViewSetDetail(TestCase):
+    client_class = APITestClient
+
+    def setUp(self):
+        self.user = user_factory()
+        self.collection = collection_factory(author=self.user)
+        self.url = reverse(
+            'collection-detail', kwargs={
+                'user_pk': self.user.pk, 'slug': self.collection.slug})
+        super(TestCollectionViewSetDetail, self).setUp()
+
+    def test_basic(self):
+        response = self.client.get(self.url)
+        assert response.status_code == 200
+        assert response.data['id'] == self.collection.id
+
+    def test_not_listed(self):
+        self.collection.update(listed=False)
+
+        response = self.client.get(self.url)
+        assert response.status_code == 401
+
+    def test_not_listed_self(self):
+        self.collection.update(listed=False)
+
+        self.client.login_api(self.user)
+        response = self.client.get(self.url)
+        assert response.status_code == 200
+        assert response.data['id'] == self.collection.id
+
+    def test_not_listed_admin(self):
+        random_user = user_factory()
+        collection = collection_factory(author=random_user, listed=False)
+
+        self.grant_permission(self.user, 'Users:Edit')
+        self.client.login_api(self.user)
+        response = self.client.get(
+            reverse('collection-detail', kwargs={'user_pk': random_user.pk,
+                                                 'slug': collection.slug}))
+        assert response.status_code == 200
+        assert response.data['id'] == collection.id
+
+    def test_disallowed_verbs(self):
+        response = self.client.post(self.url)
+        assert response.status_code == 405
+        response = self.client.put(self.url)
+        assert response.status_code == 405
+        response = self.client.patch(self.url)
+        assert response.status_code == 405
```

Comment:
Also add a couple 404 checks: if the user does not exist, if the collection does not exist.